mypyc/codegen/emitmodule.py

```diff
@@ -918,8 +921,19 @@ def generate_module_def(self, emitter: Emitter, module_name: str, module: Module
 
         emitter.emit_line('return {};'.format(module_static))
         emitter.emit_lines('fail:',
-                           '{} = NULL;'.format(module_static),
-                           'return NULL;')
+                           'Py_CLEAR({});'.format(module_static),
+                           'Py_CLEAR(modname);')
+        # the type objects returned from CPyType_FromTemplate are all new references
```

Comment:
I think this comment has drifted away from the code it is describing

Reply:
This should be fixed in ff8d287f196b98cc55809ecb72a4f4b674bc09bc.